ui/widgets/tracklist.go

```diff
@@ -70,12 +70,13 @@ type Tracklist struct {
 	DisableSorting bool
 
 	// user action callbacks
-	OnPlayTrackAt   func(int)
-	OnPlaySelection func(tracks []*mediaprovider.Track, shuffle bool)
-	OnAddToQueue    func(trackIDs []*mediaprovider.Track)
-	OnAddToPlaylist func(trackIDs []string)
-	OnSetFavorite   func(trackIDs []string, fav bool)
-	OnSetRating     func(trackIDs []string, rating int)
+	OnPlayTrackAt        func(int)
+	OnPlaySelection      func(tracks []*mediaprovider.Track, shuffle bool)
+	OnAddToQueue         func(trackIDs []*mediaprovider.Track)
+	OnAddToPlaylist      func(trackIDs []string)
+	OnSetFavorite        func(trackIDs []string, fav bool)
+	OnSetRating          func(trackIDs []string, rating int)
+	OnShowDownloadDialog func(track *mediaprovider.Track)
```

Comment:
I would change the signature of this callback to

```go
OnDownload func(trackIDs []string)
```

and in the controller, you can do the loop to show a file dialog for each track. 

Though I think it'd be a better user experience for downloading multiple tracks at once to save to a single zip file instead of showing a file chooser for each track. What do you think? Only question then would be what should be the default filename. Maybe `downloaded_tracks.zip`?

Reply:
Yes, you're right, it's a better user experience. I'll implement downloading multiple tracks into a single zip file and make the changes to the callback signature. In this case, the loop will not be needed to display the file dialog for each track. Thanks!